src/resp.rs

```diff
@@ -24,25 +24,28 @@ impl<T : Send + 'static> SingleResponse<T> {
         SingleResponse(Box::new(f))
     }
 
-    pub fn metadata_and_future<F>(metadata: Metadata, result: F) -> SingleResponse<T>
+    pub fn metadata_and_future<F>(metadata: Metadata, result: F, trailing: GrpcFutureSend<Metadata>) -> SingleResponse<T>
         where F : Future<Item=T, Error=error::Error> + Send + 'static
     {
-        let boxed: GrpcFutureSend<(T, Metadata)> = Box::new((result.map(|r| (r, Metadata::new()))));
+        let boxed: GrpcFutureSend<(T, Metadata)> = Box::new((result.map(|r| (
+            r,
+            trailing.wait().unwrap_or(Metadata::new())
+        ))));
         SingleResponse::new(future::ok((metadata, boxed)))
     }
 
-    pub fn completed_with_metadata(metadata: Metadata, r: T) -> SingleResponse<T> {
-        SingleResponse::metadata_and_future(metadata, future::ok(r))
+    pub fn completed_with_metadata(metadata: Metadata, r: T, trailing:Metadata) -> SingleResponse<T> {
+        SingleResponse::metadata_and_future(metadata, future::ok(r), future::ok(trailing).boxed())
     }
 
     pub fn completed(r: T) -> SingleResponse<T> {
-        SingleResponse::completed_with_metadata(Metadata::new(), r)
+        SingleResponse::completed_with_metadata(Metadata::new(), r, Metadata::new())
     }
 
     pub fn no_metadata<F>(r: F) -> SingleResponse<T>
         where F : Future<Item=T, Error=error::Error> + Send + 'static
     {
-        SingleResponse::metadata_and_future(Metadata::new(), r)
+        SingleResponse::metadata_and_future(Metadata::new(), r, future::ok(Metadata::new()).boxed())
```

Comment:
`.boxed()` part won't be needed if constructor has type parameter for future.

BTW, grpc-rust project uses `Box::new(future)` instead of `future.boxed()`.